objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestUploadAndDeleteQuery.java

```diff
@@ -0,0 +1,612 @@
+        checkNotUploadToVantiq(IMAGE_NAME_5);
+    }
+    
+    @Test
+    public void testInvalidImageDeleteParameters() {
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        
+        File d = new File(OUTPUT_DIR);
+        
+        // Forgetting to include imageDir
+        Map<String, Object> request = new LinkedHashMap<>();
+        request.put("imageName", "all");
+        core.deleteLocalImages(request, null); 
+        
+        File[] dList = d.listFiles();
+        assert dList.length == 5;
+        
+        // Using wrong type for imageDir
+        request.put("imageDir", 5);
+        core.deleteLocalImages(request, null);
+        
+        dList = d.listFiles();
+        assert dList.length == 5;
+        
+        
+        // Forgetting to include either imageName or imageDate
+        request.put("imageDir", OUTPUT_DIR);
+        request.remove("imageName");
+        core.deleteLocalImages(request, null);
+        
+        dList = d.listFiles();
+        assert dList.length == 5;
+        
+        
+        // Using wrong type for imageName
+        request.put("imageName", 5);
+        core.deleteLocalImages(request, null);
+        
+        dList = d.listFiles();
+        assert dList.length == 5;
+        
+        
+        // Using wrong type for imageDate
+        request.remove("imageName");
+        request.put("imageDate", 5);
+        core.deleteLocalImages(request, null);
+        
+        dList = d.listFiles();
+        assert dList.length == 5;
+    }
+    
+    @Test
+    public void testImageNameUploadAll() throws InterruptedException {
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        
+        Map<String, Object> request = new LinkedHashMap<>();
+        request.put("imageDir", OUTPUT_DIR);
+        request.put("imageName", "all");
+        
+        core.uploadLocalImages(request, null);
+                
+        // Checking that all images were uploaded to VANTIQ
+        Thread.sleep(1000);
+        checkUploadToVantiq(IMAGE_NAME_1);
+        checkUploadToVantiq(IMAGE_NAME_2);
+        checkUploadToVantiq(IMAGE_NAME_3);
+        checkUploadToVantiq(IMAGE_NAME_4);
+        checkUploadToVantiq(IMAGE_NAME_5);        
+    }
+    
+    @Test
+    public void testImageNameUploadOne() throws InterruptedException {
+        // Only run test with intended vantiq availability
+        assumeTrue(testAuthToken != null && testVantiqServer != null);
+        
+        Map<String, Object> request = new LinkedHashMap<>();
+        request.put("imageDir", OUTPUT_DIR);
+        request.put("imageName", START_DATE);
+        
+        core.uploadLocalImages(request, null);
+                
+        // Checking that all images were uploaded to VANTIQ
+        Thread.sleep(1000);
+        checkUploadToVantiq(IMAGE_NAME_2);
```

Comment:
Up to you here, but I find it useful to create a map that has all the related stuff together.  Thus, IMAGE_2 might be a map that has a file name, a date specification, etc., all in the same place.  Easier to track if/when things change later...
Optional, as I said.